cmd/templ/lspcmd/proxy/server.go

```diff
@@ -726,6 +728,74 @@ func (p *Server) DidClose(ctx context.Context, params *lsp.DidCloseTextDocumentP
 func (p *Server) DidOpen(ctx context.Context, params *lsp.DidOpenTextDocumentParams) (err error) {
 	p.Log.Info("client -> server: DidOpen", slog.String("uri", string(params.TextDocument.URI)))
 	defer p.Log.Info("client -> server: DidOpen end")
+
+	if !p.NoPreload {
+		return p.didOpen(ctx, params)
+	}
+
+	isTemplFile, goURI := convertTemplToGoURI(params.TextDocument.URI)
```

Comment:
These two lines are repeated here in and original `didOpen` down below.  You could put the NoPreload conditional after these